e2eTests/scenarios/FederatedCredentials/runtest.ps1

```diff
@@ -0,0 +1,138 @@
+$repository = 'microsoft/bcsamples-bingmaps.appsource'
+
+SetTokenAndRepository -github:$github -githubOwner $githubOwner -token $token -repository $repository
+$headers = @{
+    "Authorization" = "token $token"
+    "X-GitHub-Api-Version" = "2022-11-28"
+    "Accept" = "application/vnd.github+json"
+}
+$existingBranch = gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/$repository/branches/$branch 2> $null | ConvertFrom-Json
+if ($existingBranch.PSObject.Properties.Name -eq 'Name' -and $existingBranch.Name -eq $branch) {
+    Write-Host "Removing existing branch $branch"
+    Invoke-RestMethod -Method Delete -Uri "https://api.github.com/repos/$repository/git/refs/heads/$branch" -Headers $headers
+    Start-Sleep -Seconds 10
+}
+$latestSha = (gh api /repos/$repository/commits/main | ConvertFrom-Json).sha
+gh api --method POST -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/$repository/git/refs -f ref=refs/heads/$branch -f sha=$latestSha
+
+# Upgrade AL-Go System Files to test version
+RunUpdateAlGoSystemFiles -directCommit -wait -templateUrl $template -repository $repository -branch $branch | Out-Null
+
+# Run CI/CD workflow
+$run = RunCICD -repository $repository -branch $branch -wait
+
+# Check that workflow run uses federated credentials and signing was successful
+Test-LogContainsFromRun -repository $repository -runid $run.id -jobName 'Build Main App (Default)  Main App (Default)' -stepName 'Sign' -expectedText 'Connecting to Azure using clientId and federated token'
+Test-LogContainsFromRun -repository $repository -runid $run.id -jobName 'Build Main App (Default)  Main App (Default)' -stepName 'Sign' -expectedText 'Signing succeeded'
+
+# Check that Deliver to AppSource uses federated credentials and that a new submission was created
+Test-LogContainsFromRun -repository $repository -runid $run.id -jobName 'Deliver to AppSource' -stepName 'Read secrets' -expectedText 'Query federated token'
+Test-LogContainsFromRun -repository $repository -runid $run.id -jobName 'Deliver to AppSource' -stepName 'Deliver' -expectedText 'New AppSource submission created'
+
+# Test artifacts generated
+$artifacts = gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/$repository/actions/runs/$($run.id)/artifacts | ConvertFrom-Json
+@($artifacts.artifacts.Name) -like "Library Apps-$branch-Apps-*.*.*.0" | Should -Be $true
+@($artifacts.artifacts.Name) -like "Main App-$branch-Apps-*.*.*.0" | Should -Be $true
+@($artifacts.artifacts.Name) -like "Main App-$branch-Dependencies-*.*.*.0" | Should -Be $true
+
+Write-Host "Download build artifacts"
+invoke-gh run download $run.id --repo $repository --dir 'signedApps'
+$appFile = (Get-Item "signedApps/Main App-$branch-Apps-*.*.*.0/*.app").FullName
+
+# Check app signature
+Write-Host "Register NavSip.dll"
+Register-NavSip
+Write-Host "Check App Signature"cd
```

Comment:
```suggestion
Write-Host "Check App Signature"
```

Typo?

Reply:
Yeah - and the test code also doesn't run on ubuntu - changing that